docker/Dockerfile.firedrake

```diff
@@ -1,29 +1,30 @@
-FROM firedrakeproject/firedrake-vanilla
+FROM ghcr.io/mesh-adaptation/petsc-parmmg
 
 USER firedrake
 WORKDIR /home/firedrake
 
-# Reconfigure PETSc with packages required for mesh adaptation and rebuild
-RUN bash -c "cd petsc && \
-	  git fetch && \
-	  cp default/lib/petsc/conf/reconfigure-default.py . && \
-	  sed -i 's/]/]\n  configure_options.extend(sys.argv[1:])/g' reconfigure-default.py && \
-	  ./reconfigure-default.py $(echo -n $(curl https://raw.githubusercontent.com/mesh-adaptation/mesh-adaptation-docs/main/install/petsc_reconfigure_options.txt)) && \
-    make all"
+ENV PETSC_ARCH default
 
-# Rebuild Firedrake and install additional packages
-RUN bash -c "source firedrake/bin/activate && \
-	  cd firedrake/src/firedrake && \
-	  firedrake-update --install thetis && \
-	  cd .. && \
+# Build Firedrake and install additional packages
+RUN curl -O https://raw.githubusercontent.com/firedrakeproject/firedrake/master/scripts/firedrake-install
+RUN bash -c "python3 firedrake-install \
```

Comment:
This failed for me during `docker build`:
```
0.839 apt-get not found or disabled. Proceeding on the rash assumption that your compiled dependencies are in place.
0.839 Please use the --show-dependencies for more information.
0.839 Creating firedrake venv in '/home/firedrake/firedrake'.
20.59 BLAS config is {}
20.59 Configuration saved to /home/firedrake/firedrake/.configuration.json
20.59 Cloning firedrake
20.59 
35.05 Successfully cloned repository firedrake.
35.05 Checking out branch master
35.13 Successfully checked out branch master
35.13 Updating submodules.
35.14 Successfully updated submodules.
35.14 Creating firedrake-update script.
35.15 Pip installing VTK to venv
35.48 Traceback (most recent call last):
35.48   File "/home/firedrake/firedrake-install", line 1793, in <module>
35.48     run_pip_install_wrap(["vtk>=9.0.1"], {})
35.48   File "/home/firedrake/firedrake-install", line 1007, in run_pip_install_wrap
35.48     run_pip_install([req])
35.48   File "/home/firedrake/firedrake-install", line 991, in run_pip_install
35.48     check_call(pipinstall + pipargs)
35.48   File "/home/firedrake/firedrake-install", line 672, in check_call
35.48     log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT, env=os.environ).decode())
35.48               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
35.48   File "/usr/lib/python3.12/subprocess.py", line 466, in check_output
35.48     return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
35.48            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
35.48   File "/usr/lib/python3.12/subprocess.py", line 571, in run
35.49     raise CalledProcessError(retcode, process.args,
35.49 subprocess.CalledProcessError: Command '['/home/firedrake/firedrake/bin/python', '-m', 'pip', 'install', '--no-build-isolation', '--no-binary', 'mpi4py,randomgen,numpy', '-v', 'vtk>=9.0.1']' returned non-zero exit status 1.
```

Reply:
I then tried running the petsc container (image built from Dockerfile.petsc) and running this, but then got:
```
Running firedrake-install --no-package-manager --disable-ssh --torch --honour-petsc-dir --mpicc=/home/firedrake/petsc/packages/bin/mpicc --mpicxx=/home/firedrake/petsc/packages/bin/mpicxx --mpif90=/home/firedrake/petsc/packages/bin/mpif90 --mpiexec=/home/firedrake/petsc/packages/bin/mpiexec --install thetis
Traceback (most recent call last):
  File "/home/firedrake/firedrake-install", line 367, in <module>
    petsc_int_type = honour_petsc_dir_get_petsc_int_type()
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/firedrake/firedrake-install", line 225, in honour_petsc_dir_get_petsc_int_type
    line = honour_petsc_dir_fetch_petscconf("#define PETSC_USE_64BIT_INDICES")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/firedrake/firedrake-install", line 218, in honour_petsc_dir_fetch_petscconf
    with open(petscconf_h) as fh:
         ^^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/home/firedrake/petsc/include/petscconf.h'
```

Reply:
Huh. I did see errors with `petscconf.h` getting moved in the CI but they went away. Note the lines
```
        mv ${PACKAGES}/include/petscconf.h ${PACKAGES}/include/old_petscconf.nope;'
# Don't run make here, we only want MPICH and HWLOC
# It is also necessary to move `petscconf.h` so packages isn't treated like a working PETSc
```

When you saw this issue, what were you setting as the `PETSC_ARCH`? `default`, `packages`, or something else?

Reply:
I just tried this on my other computer (with windows) and it worked! These errors happened on my macbook M1 which is a bit notorious for being problematic - although I've never had issues with docker. And I just ran it exactly as it is, so with `PETSC_ARCH = default`.

I can investigate more if you'd like but I think it's all good :)

Reply:
Hm okay. Thanks for checking! I opened issue https://github.com/mesh-adaptation/mesh-adaptation-docs/issues/44 and will go ahead to merge this.